src/traces/requestProcessor.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Context, SpanKind, TraceFlags } from "@opentelemetry/api";
+import { ReadableSpan, Span, SpanProcessor } from "@opentelemetry/sdk-trace-base";
+
+/**
+ * Azure Monitor Incoming & Outgoing Request Processor.
+ * @internal
+ */
+export class RequestSpanProcessor implements SpanProcessor {
+    private _enableDependencyTelemetry: boolean;
+    private _enableRequestTelemetry: boolean;
+
+    constructor(enableDependencyTelemetry: boolean, enableRequestTelemetry: boolean) {
+        this._enableDependencyTelemetry = enableDependencyTelemetry;
+        this._enableRequestTelemetry = enableRequestTelemetry;
+    }
+
+    forceFlush(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    onStart(span: Span, _context: Context): void {
+        return;
+    }
+
+    onEnd(span: ReadableSpan): void {
```

Comment:
Based on the specification you are "not allowed" to alter the span or the span context in any way during the `onEnd` as this is "supposed" to be a readable span (not a mutable one).

There is a new `onEnding` which receives a "Mutable" Span to allow the setting / changing like this -- not sure if OTel-JS has this yet

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#onending

Reply:
While `onEnding` doesn't appear to be implemented on the JS side, I've implemented and tested this logic in the `onStart()` method with good results.